File: 0070-climbing-stairs/0070-climbing-stairs.rb
# @return {Integer}
def climb_stairs(n)
  return 1 if n == 1
  return 2 if n == 2
  fb = []
  fb[1] = 1
  fb[2] = 2
  3.upto(n) do |x|
    fb[x] = fb[x-1] + fb[x-2]
  end
  fb.last
end

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 2:
            return n
        
        dp = [1] * (n+1)
        
        for i in range(2,n+1):
            dp[i] = dp[i-1] + dp[i-2]
        
        return dp[n]